pkg/registry/servicecatalog/broker/strategy.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
+
+// this was copied from where else and edited to fit our objects
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+/* Begin Create Definition */
+
+type brokerCreateStrategy struct {
+	runtime.ObjectTyper // inherit ObjectKinds method
+	kapi.NameGenerator  // GenerateName method for CreateStrategy
+}
+
+// implements RESTCreateStrategy interface
+var createStrategy = brokerCreateStrategy{
+	// embeds to pull in existing code behavior from upstream
+
+	// this has an interesting NOTE on it. Not sure if it applies to us.
```

Comment:
I think we're good until the work is finished and we re-vendor. We may have new dependencies on `kubernetes/apimachinery` repo as well.
```
pmorie [1:23 PM] 
okay, so here's a question @claytonc @deads2k -- there are some comments that indicate that people should implement their own API schemes, which seems wrong to me.  is that actually the recommendation?  it seems like api.Scheme's tendrils run far and deep to me. (edited)

[1:23]  
cc @mbauer

deads2k [1:26 PM] 
@pmorie they should.  api.scheme is going to be disconnected from genericapiserver and you'll have to pass your scheme in
```